# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [1]:
import pandas as pd
import numpy as np

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [2]:
first = pd.Series(np.random.randint(low=1, high=4, size=100))
second = pd.Series(np.random.randint(low=1, high=3, size=100))
third = pd.Series(np.random.randint(low=1000, high=30000, size=100))

### Step 3. Let's create a DataFrame by joinning the Series by column

In [6]:
df = pd.concat([first, second, third], axis=1)
df

0  1      2
0   3  1  21893
1   1  2  13824
2   3  1   1265
3   3  2  29091
4   2  2  16291
.. .. ..    ...
95  2  1  12627
96  1  2  11597
97  1  1  17818
98  1  2   5114
99  3  1  20119

[100 rows x 3 columns]

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [9]:
columns = ['bedrs', 'bathrs', 'price_sqr_meter']
df = df.rename(columns=dict(zip(df.columns, columns)))
df

bedrs  bathrs  price_sqr_meter
0       3       1            21893
1       1       2            13824
2       3       1             1265
3       3       2            29091
4       2       2            16291
..    ...     ...              ...
95      2       1            12627
96      1       2            11597
97      1       1            17818
98      1       2             5114
99      3       1            20119

[100 rows x 3 columns]

### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [16]:
bigcolumn = pd.concat([first, second, third], ignore_index=True)
bigcolumn = bigcolumn.to_frame()
bigcolumn

0
0        3
1        1
2        3
3        3
4        2
..     ...
295  12627
296  11597
297  17818
298   5114
299  20119

[300 rows x 1 columns]

### Step 6. Oops, it seems it is going only until index 99. Is it true?

### Step 7. Reindex the DataFrame so it goes from 0 to 299